### ISSUE TITLE
lint: consider to use regexp.MustCompile instead of regexp.Compile

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#209]
Name: `regexpChecker`
before:
```go
re, err := regexp.Compile("abc(d|e)") // const pattern
```
after:
```go
re := regexp.MustCompile("abc(d|e)")
```
From: https://github.com/cristaloleg/go-advices

**Comment by Quasilyte** [comment]
Makes sense. I don't think case with incorrect hardcoded pattern requires any "error handling".
Should be always valid (unless I'm missing something).

**Comment by cristaloleg** [comment]
'Cause regexp is const it can be manually verified is it correct before using it in code.
I don't see any corner cases to be honest.

**Comment by cristaloleg** [comment]
Awesome! we can really easily validate user's regex during linter's run.

**Comment by Quasilyte** [comment]
Megacheck also does html/template text/template validation. :)
Though we might investigate whether it's possible to suggest `template.Must`.
